Cannot find module '@vue/cli-plugin-router/generator/template/src/views/Home.vue'

### Version
4.0.0-alpha.5

### Reproduction link
[https://github.com/vuejs/vue-cli/issues/new](https://github.com/vuejs/vue-cli/issues/new)





### Environment info
```
Environment Info:

  System:
    OS: macOS 10.14.5
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  Binaries:
    Node: 12.4.0 - /usr/local/bin/node
    Yarn: 1.17.0 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 76.0.3809.62
    Firefox: 65.0.2
    Safari: 12.1.1
  npmGlobalPackages:
    @vue/cli: Not Found
```


### Steps to reproduce
vue create sample
typescript

### What is expected?
create success

### What is actually happening?
Error: Cannot find module '@vue/cli-plugin-router/generator/template/src/views/Home.vue' from '/Users/xiaoyu/wwwroot/node/neoworld4/node_modules/@vue/cli-plugin-typescript/generator/template/src/views'
    at Function.module.exports [as sync] (/Users/xiaoyu/.config/yarn/global/node_modules/resolve/lib/sync.js:71:15)
    at renderFile (/Users/xiaoyu/.config/yarn/global/node_modules/@vue/cli/lib/GeneratorAPI.js:408:17)
    at /Users/xiaoyu/.config/yarn/global/node_modules/@vue/cli/lib/GeneratorAPI.js:244:27
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
    at async Generator.resolveFiles (/Users/xiaoyu/.config/yarn/global/node_modules/@vue/cli/lib/Generator.js:209:7)
    at async Generator.generate (/Users/xiaoyu/.config/yarn/global/node_modules/@vue/cli/lib/Generator.js:116:5)
    at async Creator.create (/Users/xiaoyu/.config/yarn/global/node_modules/@vue/cli/lib/Creator.js:192:5)
    at async create (/Users/xiaoyu/.config/yarn/global/node_modules/@vue/cli/lib/create.js:73:3)

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/packages/@vue/cli/lib/GeneratorAPI.js
+++ b/packages/@vue/cli/lib/GeneratorAPI.js
@@ -417,6 +417,22 @@ function renderFile (name, data, ejsOptions) {
   const parsed = yaml.loadFront(template)
   const content = parsed.__content
   let finalTemplate = content.trim() + `\n`
+
+  if (parsed.when) {
+    finalTemplate = (
+      `<%_ if (${parsed.when}) { _%>` +
+        finalTemplate +
+      `<%_ } _%>`
+    )
+
+    // use ejs.render to test the conditional expression
+    // if evaluated to falsy vaule, return early to avoid extra cost for extend expression
+    const result = ejs.render(finalTemplate, data, ejsOptions)
+    if (!result) {
+      return
+    }
+  }
+
   if (parsed.extend) {
     const extendPath = path.isAbsolute(parsed.extend)
       ? parsed.extend
@@ -436,13 +452,6 @@ function renderFile (name, data, ejsOptions) {
       } else {
         finalTemplate = finalTemplate.replace(parsed.replace, content.trim())
       }
-    }
-    if (parsed.when) {
-      finalTemplate = (
-        `<%_ if (${parsed.when}) { _%>` +
-          finalTemplate +
-        `<%_ } _%>`
-      )
     }
   }
 

